### PR TITLE
Initial discord bot logic

### PR DESCRIPTION
**Pull request by PauliusLozys**
This adds initial discord bot logic code. Further PRs will expands its functionality.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	discord.AddHandler(func(s *discordgo.Session, m *discordgo.MessageCreate) {
+		if m.Author.ID == s.State.User.ID { // Ignore your own messages.
+			return
+		}
+
+		// TODO: Message handling logic.
```
Comment (domgud) [inline cmd/main.go:29]: TODO should have a ticket reference 🐱‍👤

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+go 1.21
+
+require (
+	github.com/bwmarrin/discordgo v0.27.1
+	github.com/joho/godotenv v1.5.1
+)
+
+require (
+	github.com/gorilla/websocket v1.4.2 // indirect
```
Comment (domgud) [inline go.mod:11]: Is there any specific reason to use `1.4.2` I can see that there are newer versions available such as `1.5.1`?

File: go.sum [hunk 3]
```diff
@@ -0,0 +1,14 @@
+github.com/bwmarrin/discordgo v0.27.1 h1:ib9AIc/dom1E/fSIulrBwnez0CToJE113ZGt4HoliGY=
+github.com/bwmarrin/discordgo v0.27.1/go.mod h1:NJZpH+1AfhIcyQsPeuBKsUtYrRnjkyu0kIVMCHkZtRY=
+github.com/gorilla/websocket v1.4.2 h1:+/TMaTYc4QFitKJxsQ7Yye35DkWvkdLcvGKqM+x0Ufc=
```
Comment (domgud) [inline go.sum:3]: Same comment about versions

File: go.mod (hunk 2 above)
Comment (PauliusLozys) [reply go.mod:11]: This is automatically managed by go package manager. As the comment suggest, this is an indirect dependency by `discordgo` package itself and we cannot control it.

File: go.sum (hunk 3 above)
Comment (PauliusLozys) [reply go.sum:3]: `go.sum` is meant as checksum for `go.mod` we usually don't control the content of it.

File: go.mod (hunk 2 above)
Comment (domgud) [reply go.mod:11]: Cringe 😿

File: cmd/main.go (hunk 1 above)
Comment (PauliusLozys) [reply cmd/main.go:29]: Made issue to address this: https://github.com/PauliusLozys/gamba-bot/issues/2

**Comment by domgud** [comment]
LGTM!

**Review by domgud — APPROVED** [review]

**Pull request merged**